provider/execution/execution.go

```diff
@@ -0,0 +1,371 @@
+		resource := schema.NewResourceData(e.Db.Dialect(), e.Table, parent, o, e.extraFields, e.executionStart)
+		// Before inserting resolve all table column resolvers
+		if err := e.resolveResourceValues(ctx, meta, resource); err != nil {
+			e.Logger.Warn("skipping failed resolved resource", "reason", err.Error())
+			diags = diags.Append(err)
+			continue
+		}
+		resources = append(resources, resource)
+	}
+
+	// only top level tables should cascade
+	shouldCascade := parent == nil
+	resources, dbDiags := e.copyDataIntoDB(ctx, resources, shouldCascade)
+	diags = diags.Append(dbDiags)
+	totalCount := uint64(len(resources))
+
+	// Finally, resolve relations of each resource
+	for _, rel := range e.Table.Relations {
+		meta.Logger().Debug("resolving table relation", "table", e.Table.Name, "relation", rel.Name)
+		for _, r := range resources {
+			// ignore relation resource count
+			if _, innerDiags := e.WithTable(rel).Resolve(ctx, meta, r); innerDiags.HasDiags() {
+				diags = diags.Append(innerDiags)
+			}
+		}
+	}
+	return totalCount, diags
+}
+
+func (e *TableExecutor) copyDataIntoDB(ctx context.Context, resources schema.Resources, shouldCascade bool) (schema.Resources, diag.Diagnostics) {
+	err := e.Db.CopyFrom(ctx, resources, shouldCascade, e.extraFields)
+	if err == nil {
+		return resources, nil
+	}
+	e.Logger.Warn("failed copy-from to db", "error", err, "table", e.Table.Name)
+
+	// fallback insert, copy from sometimes does problems, so we fall back with bulk insert
+	err = e.Db.Insert(ctx, e.Table, resources)
+	if err == nil {
+		return resources, nil
+	}
+	e.Logger.Error("failed insert to db", "error", err, "table", e.Table.Name)
+
+	// Setup diags, adding first diagnostic that bulk insert failed
+	diags := diag.Diagnostics{}.Append(FromError(err, WithType(diag.DATABASE), WithSummary("failed bulk insert on table %s", e.Table.Name)))
+	// Try to insert resource by resource if partial fetch is enabled and an error occurred
+	partialFetchResources := make(schema.Resources, 0)
+	for id := range resources {
+		if err := e.Db.Insert(ctx, e.Table, schema.Resources{resources[id]}); err != nil {
+			e.Logger.Error("failed to insert resource into db", "error", err, "resource_keys", resources[id].PrimaryKeyValues(), "table", e.Table.Name)
+			diags = diags.Append(FromError(err, WithType(diag.DATABASE), WithErrorClassifier))
+			continue
+		}
+		// If there is no error we add the resource to the final result
+		partialFetchResources = append(partialFetchResources, resources[id])
+	}
+	return partialFetchResources, diags
+}
+
+func (e *TableExecutor) resolveResourceValues(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource) (err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			e.Logger.Error("resolve resource recovered from panic", "table", e.Table.Name, "stack", string(debug.Stack()))
+			err = FromError(fmt.Errorf("column resolve panic: %s", r), WithResource(e.ResourceName), WithSeverity(diag.PANIC),
```

Comment:
Same panic comments here, where would the stack trace go?

Reply:
Added stack as details